lua/core/mappings.lua

```diff
@@ -96,73 +96,79 @@ M.comment = function()
    map("v", "<leader>/", "<esc><cmd> :lua require('Comment.api').toggle_linewise_op(vim.fn.visualmode())<CR>")
 end
 
-M.lspconfig = function()
+M.lspconfig = function(bufnr)
    -- See `<cmd> :help vim.lsp.*` for documentation on any of the below functions
-   map("n", "gD", function()
+
+   local buf_map = function(...)
+      local key, lhs, rhs = ...
```

Comment:
This probably can be simplified with Lua trick that I don't know

Reply:
no worries, its alright for the time being